src/main/java/com/google/cloud/spanner/pgadapter/utils/Metrics.java

```diff
@@ -31,31 +30,42 @@ public class Metrics {
   static final String PGADAPTER_LATENCY_DESCRIPTION =
       "Latency between PGAdapter receiving a statement from the client and PGAdapter returning the last row of the response to the client";
 
-  private final LongHistogram spannerClientLibLatencies;
-  private final LongHistogram pgadapterLatencies;
+  private final DoubleHistogram spannerClientLibLatencies;
+  private final DoubleHistogram pgadapterLatencies;
+
+  public static List<Double> getMetricLatencyMillisBuckets() {
+    int kMaxNumFiniteBuckets = 50;
+    double BASE = 1.25;
+    double SCALE_FACTOR = 0.25;
+    double MAX_VALUE = 600;
+    double bucket_value = SCALE_FACTOR;
+
+    List<Double> rpc_millis_bucket_boundaries = new ArrayList<Double>(kMaxNumFiniteBuckets);
+    for (int i = 0; i < kMaxNumFiniteBuckets; i++) {
+      if (bucket_value > MAX_VALUE) {
+        break;
+      }
```

Comment:
nit: you could move this check into the for loop like this:

```suggestion
    for (int i = 0; i < kMaxNumFiniteBuckets && bucket_value <= MAX_VALUE; i++) {
      if (bucket_value > MAX_VALUE) {
        break;
      }
```